src/site/content/en/blog/preload-scanner/index.md

```diff
@@ -0,0 +1,285 @@
+---
+title: Don't fight the browser preload scanner
+subhead: |
+  Find out what the browser preload scanner is, how it helps performance, and how you can stay out of its way.
+authors:
+  - jlwagner
+date: 2022-05-13
+hero: image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/ZYfhboCPqPWawQ5LzWKO.jpg
+thumbnail: image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/irM6kTXZoP0MmXGtHOSa.jpg
+alt: A close-up photograph of a screen with various bits of HTML markup shown in a text editor.
+description: |
+  Find out what the browser preload scanner is, how it helps performance, and how you can stay out of its way.
+tags:
+  - blog
+  - performance
+  - web-vitals
+---
+
+One overlooked aspect of optimizing page speed involves knowing a bit about browser internals. Browsers make certain optimizations for us to improve performance in ways that we as developers can't—but only so long as we don't thwart those optimizations unintentionally.
+
+One internal browser optimization to understand is the browser preload scanner. In this post, we'll talk a bit about how the preload scanner works—and more importantly, how you can avoid getting in its way.
+
+## What's a preload scanner?
+
+Every browser has a primary HTML parser that [tokenizes](https://en.wikipedia.org/wiki/Lexical_analysis#Tokenization) raw markup and processes it into [an object model](https://developer.mozilla.org/docs/Web/API/Document_Object_Model). This all merrily goes on until the parser pauses when it finds a [blocking resource](/render-blocking-resources/), such as a stylesheet loaded with a `<link>` element, or script loaded with a `<script>` element without an `async` or `defer` attribute.
+
+<figure>
+  {% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/mXRoJneD6CbMAqaTqNZW.svg", alt="A diagram of how the browser's primary HTML parser parses and tokenizes markup into the DOM. The parser is blocked from doing any further work when it encounters a &lt;link&gt; element pointing to a stylesheet, and can only resume its work until the stylesheet is downloaded and processed.", width="111", height="150" %}
+  <figcaption>
+    <strong>Fig. 1:</strong> A diagram of how the browser's primary HTML parser can be blocked. In this case, the parser runs into a <code>&lt;link&gt;</code> element for an external CSS file, which blocks the browser from parsing the rest of the document—or even rendering any of it—until the CSS is downloaded and parsed.
+  </figcaption>
+</figure>
```

Comment:
Note on a11y: A screenreader would read the alt text followed by the caption, thus repeating the info. You have a great caption for this image, so let it do the heavy lifting. I'll make a suggestion for lighter alt text.

Reply:
```suggestion
<figure>
  {% Img src="image/jL3OLOhcWUQDnR4XjewLBx4e3PC3/mXRoJneD6CbMAqaTqNZW.svg", alt="HTML parser diagram.", width="111", height="150" %}
  <figcaption>
    <strong>Fig. 1:</strong> A diagram of how the browser's primary HTML parser can be blocked. In this case, the parser runs into a <code>&lt;link&gt;</code> element for an external CSS file, which blocks the browser from parsing the rest of the document—or even rendering any of it—until the CSS is downloaded and parsed.
  </figcaption>
</figure>
```